gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -266,24 +303,29 @@ def run_translation_averaging(
 
         # Run the optimizer
         # TODO(akshay-krishnan): remove once latest gtsam pip wheels updated.
-        try:
-            algorithm = TranslationRecovery()
-            logger.debug("[1dsfm] Constructed NEW TranslationRecovery, about to run.")
-            w_i2ti1_priors = self._get_prior_measurements_in_world_frame(relative_pose_priors, wRi_list)
-            wti_initial = self.__get_initial_values(absolute_pose_priors)
-            logger.debug("[1dsfm] Computed priors and initial values.")
-            if len(w_i2ti1_priors) > 0:
-                # scale is ignored here.
-                wti_values = algorithm.run(w_i2Ui1_measurements, 0.0, w_i2ti1_priors, wti_initial)
-                logger.debug("[1dsfm] Finished with priors.")
-            else:
-                wti_values = algorithm.run(w_i2Ui1_measurements, scale_factor)
-                logger.debug("[1dsfm] Finished without priors.")
-        except TypeError:
-            recovery = TranslationRecovery(w_i2Ui1_measurements)
-            logger.debug("[1dsfm] Constructed OLD TranslationRecovery, about to run.")
-            wti_values = recovery.run(scale_factor)
-            logger.debug("[1dsfm] Finished.")
+        # try:
+        algorithm = TranslationRecovery()
+        logger.debug("[1dsfm] Constructed NEW TranslationRecovery, about to run.")
+        w_relative_pose_priors = self._get_prior_measurements_in_world_frame(relative_pose_priors, wRi_list)
+        wti_initial = self.__get_initial_values(absolute_pose_priors)
+        logger.debug("[1dsfm] Computed priors and initial values.")
+        if len(w_relative_pose_priors) > 0:
+            # scale is ignored here.
+            noise_model_for_augmentation = gtsam.noiseModel.Isotropic.Sigma(NOISE_MODEL_DIMENSION, NOISE_MODEL_SIGMA)
+            augmented_w_i2Ui1_measurements = self.augment(
+                w_i2Ui1_measurements, w_relative_pose_priors, noise_model_for_augmentation
+            )
+            wti_values = algorithm.run(augmented_w_i2Ui1_measurements, 0.0, w_relative_pose_priors, wti_initial)
+            logger.debug("[1dsfm] Finished with priors.")
+        else:
+            wti_values = algorithm.run(w_i2Ui1_measurements, scale_factor)
+            logger.debug("[1dsfm] Finished without priors.")
+        # except TypeError as te:
+        #     logger.error(te)
+        #     recovery = TranslationRecovery(w_i2Ui1_measurements)
+        #     logger.debug("[1dsfm] Constructed OLD TranslationRecovery, about to run.")
+        #     wti_values = recovery.run(scale_factor)
+        #     logger.debug("[1dsfm] Finished.")
```

Comment:
@akshay-krishnan are we good with removing the try-catch block?

Reply:
Yes